articles/040_stories.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: default
+title: Stories driving the development
+permalink: articles/stories.html
+abstract:
+  This article captures some stories which drive the direction of features in ROS.
+published: true
+author: '[Dirk Thomas](https://github.com/dirk-thomas)'
+categories: Overview
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+The article enumerates a few stories which sketch what will be possible with ROS in the future.
+The list is by no means exhaustive.
+
+## Push ROS into the hardware nodes
+
+A couple of design decisions in ROS 1 make it difficult to integrate hardware devices (e.g. a sensor or actuator) natively into the ROS graph.
+The master is a central entity in the ROS graph and needs to be started before any of the nodes.
+Also the communication between the nodes and the master is done using XML-RPC which poses a significant dependency when being implemented on small resource constraint systems / micro controllers.
+Instead commonly a driver is being used which uses a custom protocol to communicate between the device and the computer and exposes a ROS interface on the computer.
+
+In the future it should be possible to implement the ROS protocol directly on the devices embedded system.
+A ROS-enabled device would then be able to discover other nodes automatically and expose a ROS interface (composed of publisher, services and parameters).
+The adoption of an industry standard like DDS as well as the decentralized nature of the middleware are important pieces to enable this.
+
+The advantage of this approach is twofold:
+
+- The integration effort to build a new system is being reduced since no custom drivers are required anymore.
+- By implementing a specific ROS-based interface the devices of specific class can easily be substituted without the need to spend time on integrating the software / hardware from e.g. a different vendor.
+
+## Delay decision on process layout to deploy time
+
+In ROS 1 nodes commonly use the `Node` API and implement their own `main` function.
+Only a few packages leverage the `Nodelet` API and compile their components into shared libraries instead.
```

Comment:
Maybe it's obvious, but you could point out that whether or not you use the Nodelet API is a programming time decision, and that the Nodelet API was different enough from the Node API to make porting a pain.